benches/server.rs

```diff
@@ -86,15 +114,37 @@ impl Service for Hello {
     }
 }
 
-fn spawn_hello(until: oneshot::Receiver<()>) -> SocketAddr {
-    let (addr_tx, addr_rx) = mpsc::channel();
-    ::std::thread::spawn(move || {
-        let addr = "127.0.0.1:0".parse().unwrap();
-        let srv = hyper::server::Http::new().bind(&addr, || Ok(Hello)).unwrap();
-        let addr = srv.local_addr().unwrap();
-        addr_tx.send(addr).unwrap();
-        srv.run_until(until.map_err(|_| ())).unwrap();
-    });
+struct ChunkedSmallPayload;
 
-    addr_rx.recv().unwrap()
+impl Service for ChunkedSmallPayload {
+    type Request = server::Request;
+    type Response = server::Response;
+    type Error = hyper::Error;
+    type Future = ::futures::Finished<Self::Response, hyper::Error>;
+    fn call(&self, _req: Self::Request) -> Self::Future {
+        ::futures::finished(
+            server::Response::new()
+                .with_header(TransferEncoding::chunked())
+                .with_header(ContentType::plaintext())
+                .with_body("Hello, World!")
+        )
+    }
 }
+
+struct ChunkedLargePayload;
+
+impl Service for ChunkedLargePayload {
+    type Request = server::Request;
+    type Response = server::Response;
+    type Error = hyper::Error;
+    type Future = ::futures::Finished<Self::Response, hyper::Error>;
+    fn call(&self, _req: Self::Request) -> Self::Future {
+        ::futures::finished(
+            server::Response::new()
+                .with_header(TransferEncoding::chunked())
+                .with_header(ContentType::plaintext())
+                .with_body((0..1_000_000).flat_map(|_| "Hello, World!".bytes()).collect::<Vec<u8>>())
```

Comment:
Oh wait, this is sending a single `Vec<u8>`, with `b"Hello, World"` in it a million times. I misread, and assumed it was sending `b"Hello, World"` as a million different chunks, and so was thinking that every single message would wrapped by the chunked header and trailer.